### PR TITLE
Add vendor files

### PR DESCRIPTION
**Pull request by shague**
Only review the last patch as this PR is based off other PRs under review: https://github.com/nimbess/stargazer/commit/fe5412ad44acf1d233d495fd51805d875511881a

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  - go get -u golang.org/x/lint/golint
+
+before_script:
+  - set -e
+  - go list ./... | grep -v /proto/ | xargs -n 1 golint
+  - go tool vet  ./cmd/ ./pkg/
+  - git ls-files | grep -v proto |grep ".go$" | xargs gofmt -l | wc -l
+
+script:
+  - echo 0
```
Comment (trozet) [inline .travis.yml:17]: do you want to add uploading the docker image on master branch merge? Can use the credentials and stuff from here:
https://github.com/nimbess/nimbess-agent/blob/master/.travis.yml

I created the nimbess/stargazer repo for you in docker

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+RUN go mod verify
+
+# Build the binary
+FROM builder-base AS builder
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
+    go build -v -o /go/bin/stargazer -ldflags "-X main.VERSION=0.0.1" -a -installsuffix cgo ./cmd/stargazer/
+
+# Copy the binary
+FROM alpine
```
Comment (trozet) [inline Dockerfile:18]: centos?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Get dependencies
+WORKDIR /go/src/github.com/nimbess/stargazer
+ENV GO111MODULE=on
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+RUN go mod verify
+
+# Build the binary
+FROM builder-base AS builder
```
Comment (trozet) [inline Dockerfile:12]: what's the point of doing this and copying everything again?

File: .gitignore [hunk 4]
```diff
@@ -11,5 +11,8 @@
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 
+# Build binary
+bin*
+
```
Comment (trozet) [inline .gitignore:16]: Why don't you add vendor files here and lets not include that code in the repo. You are already using go modules which will handle downloading dependencies (as you do in your Dockerfile).

**Review by trozet — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (shague) [reply .travis.yml:17]: yes, will add then in a follow up patch. I think this image will change a bit once I have the etcd parts included.

File: Dockerfile (hunk 3 above)
Comment (shague) [reply Dockerfile:12]: This should be a copy if I understand the docker build correctly. It is an incremental build and is suggested as being better performance.

File: .gitignore (hunk 4 above)
Comment (shague) [reply .gitignore:16]: yes, agreed. I saw other commits from others that included the vendor dir so I followed that pattern. As you mention the go modules gets around that need. I will submit another pr that removes the vendor stuff. I have other outstanding pr's and it was difficult to remove vendor dir without causing conflicts.

**Pull request merged**